src/frontend/src/icp/utils/ic-wallet.utils.ts

```diff
@@ -14,3 +23,60 @@ export const waitAndTriggerWallet = async () => {
 	// In case the best case scenario was not met, we optimistically try to retrieve the transactions on more time given that we generally retrieve transactions every WALLET_TIMER_INTERVAL_MILLIS seconds without blocking the UI.
 	waitForMilliseconds(INDEX_RELOAD_DELAY).then(() => emit({ message: 'oisyTriggerWallet' }));
 };
+
+/**
+ * Clean the workers that are not used anymore based on a given list of tokens.
+ *
+ * @param workers The map of workers defined as `Map<TokenId, WalletWorker>`.
+ * @param tokens The list of tokens to check against the workers.
+ */
+export const cleanWorkers = ({
+	workers,
+	tokens
+}: {
+	workers: Map<TokenId, WalletWorker>;
+	tokens: IcToken[];
+}) =>
+	Array.from(workers.keys())
+		.filter((tokenId) => !tokens.some(({ id: icTokenId }) => icTokenId === tokenId))
+		.forEach((tokenId) => {
+			workers.get(tokenId)?.stop();
+			workers.delete(tokenId);
```

Comment:
I'm not against a more functional programming approach. Deleting entries from a list which is the same list that is used to loop the entries isn't my favorite pattern. This can lead to some unexpected behavior generally speaking.

WDYT?

Reply:
maybe `cleanWorkers` can return a new map with the remaining `workers`?

```typescript
export const cleanWorkers = ({
	workers,
	tokens
}: {
	workers: Map<TokenId, WalletWorker>;
	tokens: IcToken[];
}): Map<TokenId, WalletWorker> =>
	new Map(
		Array.from(workers.entries()).filter(([tokenId, worker]) => {
			const isStillUsed = tokens.some(({ id: icTokenId }) => icTokenId === tokenId);
			if (!isStillUsed) {
				worker.stop();
			}
			return isStillUsed;
		})
	);
```

Reply:
Maybe add a todo for our future self?

Reply:
done